### PR TITLE
:running: followup from #1096

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: controllers/machine_controller.go [hunk 1]
```diff
@@ -297,24 +296,30 @@ func (r *MachineReconciler) drainNode(ctx context.Context, cluster *clusterv1.Cl
...
-		DryRun:                false,
+		// If a pod is not evicted in 20 seconds, retry the eviction next time the
+		// machine gets reconciled again (to allow other machines to be reconciled).
+		Timeout: 20 * time.Second,
+		OnPodDeletedOrEvicted: func(pod *corev1.Pod, usingEviction bool) {
+			verbStr := "Deleted"
+			if usingEviction {
+				verbStr = "Evicted"
+			}
+			klog.Infof("Machine %v: %s pod %s/%s from Node %q \n", pod.Namespace, pod.Name, verbStr, nodeName)
```
Comment (detiber) [inline controllers/machine_controller.go]:
```suggestion
			klog.Infof("Machine %s/%s: %s pod %s/%s from Node %q \n", cluster.Namespace, machineName, verbStr, pod.Namespace, pod.Name, nodeName)
```

File: controllers/machine_controller.go [hunk 2]
```diff
@@ -297,24 +296,30 @@ func (r *MachineReconciler) drainNode(ctx context.Context, cluster *clusterv1.Cl
...
+		Out:    writer{klog.Info},
+		ErrOut: writer{klog.Error},
+		DryRun: false,
 	}
 
 	if err := kubedrain.RunCordonOrUncordon(drainer, node, true); err != nil {
-		// Machine still tries to terminate after drain failure
-		klog.Errorf("%q: node cordon failed %q; %v", cluster.Name, nodeName, err)
+		// Machine will be re-reconciled after a cordon failure.
+		klog.Errorf("Machine %v: Cordon failed for node %q: %v", machineName, nodeName, err)
```
Comment (detiber) [inline controllers/machine_controller.go]:
```suggestion
		klog.Errorf("Machine %s/%s: Cordon failed for node %q: %v", cluster.Namespace, machineName, nodeName, err)
```

File: controllers/machine_controller.go [hunk 3]
```diff
@@ -297,24 +296,30 @@ func (r *MachineReconciler) drainNode(ctx context.Context, cluster *clusterv1.Cl
...
 		return errors.Errorf("unable to cordon node %s: %v", node.Name, err)
 	}
 
 	if err := kubedrain.RunNodeDrain(drainer, node.Name); err != nil {
-		// Machine still tries to terminate after drain failure
-		klog.Warningf("%q: drain failed for node %q; %v", cluster.Name, nodeName, err)
+		// Machine will be re-reconciled after a drain failure.
+		klog.Warningf("Machine %v: Drain failed for node %q: %v", machineName, nodeName, err)
```
Comment (detiber) [inline controllers/machine_controller.go]:
```suggestion
		klog.Warningf("Machine %s/%s: Drain failed for node %q: %v", cluster.Namespace, machineName, nodeName, err)
```

File: controllers/machine_controller.go [hunk 4]
```diff
@@ -196,7 +196,7 @@ func (r *MachineReconciler) reconcileDelete(ctx context.Context, cluster *cluste
 				r.recorder.Eventf(m, corev1.EventTypeWarning, "FailedDrainNode", "error draining Machine's node: %v", err)
 				return ctrl.Result{}, err
 			}
-			r.recorder.Eventf(m, corev1.EventTypeNormal, "SuccessfulDrainNode", "success draining Machine %q node %q", m.Name, m.Status.NodeRef.Name)
+			r.recorder.Eventf(m, corev1.EventTypeNormal, "SuccessfulDrainNode", "success draining Machine's node %q", m.Status.NodeRef.Name)
 		}
 		klog.Infof("Deleting node %q for machine %q", m.Status.NodeRef.Name, m.Name)
```
Comment (detiber) [inline controllers/machine_controller.go]:
```suggestion
		klog.Infof("Deleting node %q for machine %s/%s", m.Status.NodeRef.Name, m.Namespace, m.Name)
```

File: controllers/machine_controller.go [hunk 5]
```diff
@@ -285,7 +284,8 @@ func (r *MachineReconciler) drainNode(ctx context.Context, cluster *clusterv1.Cl
 	if err != nil {
 		if apierrors.IsNotFound(err) {
 			// If an admin deletes the node directly, we'll end up here.
-			klog.Infof("Machine %v: Could not find node %v from noderef, it may have already been deleted: %v", machineName, nodeName, cluster.Name)
+			klog.Infof("Machine %s/%s: Could not find node %v from noderef, it may have already been deleted: %v",
+				cluster.Namespace, machineName, nodeName, cluster.Name)
```
Comment (detiber) [inline controllers/machine_controller.go]: Cluster.Name feels a bit odd here, it looks like this should be `err` instead based on the use in the string above.

**Comment by vincepri** [comment]
@detiber ptal

**Pull request merged**